Sources/Honeycomb/HoneycombNavigationSpanInstrumentation.swift

```diff
@@ -0,0 +1,115 @@
+private let navigationSpanName = "Navigation"
+private let unencodablePath = "<unencodable path>"
+
+func getTracer() -> Tracer {
+    return OpenTelemetry.instance.tracerProvider.get(
+        instrumentationName: navigationInstrumentationName,
+        instrumentationVersion: honeycombLibraryVersion
+    )
+}
+
+internal class HoneycombNavigationProcessor {
+    static let singleton = HoneycombNavigationProcessor()
+    var currentNavigationPath: String? = nil
+
+    private init() {}
+
+    @available(iOS 16.0, macOS 12.0, *)
+    func reportNavigation(path: NavigationPath) {
+        if let codablePath = path.codable {
+            reportNavigation(path: codablePath)
+        } else {
+            reportNavigation(path: unencodablePath)
+        }
+    }
+
+    func reportNavigation(path: String) {
+        print("current path: \(path)")
+
+        currentNavigationPath = path
+
+        // emit a span that says we've navigated to this path
+        getTracer().spanBuilder(spanName: navigationSpanName)
+            .setAttribute(key: "NavigationPath", value: path)
+            .startSpan()
+            .end()
+    }
+
+    func reportNavigation(path: Encodable) {
+        do {
+            let encoder = JSONEncoder()
+            let data = try encoder.encode(path)
+            let pathStr = String(decoding: data, as: UTF8.self)
+
+            reportNavigation(path: pathStr)
+        } catch {
+            reportNavigation(path: unencodablePath)
+        }
+    }
+
+    func reportNavigation(path: [Encodable]) {
+        do {
+            let encoder = JSONEncoder()
+
+            let pathStr =
+                try path.map {
+                    let encoded = try encoder.encode($0)
+                    return String(decoding: encoded, as: UTF8.self)
+                }
+                .joined(separator: ",")
+
+            reportNavigation(path: "[\(pathStr)]")
+        } catch {
+            reportNavigation(path: unencodablePath)
+        }
+    }
+
+    func reportNavigation(path: Any) {
+        reportNavigation(path: unencodablePath)
+    }
+
+}
+
+extension View {
+    @available(iOS 16.0, macOS 12.0, *)
+    public func instrumentNavigation(path: NavigationPath) -> some View {
+        HoneycombNavigationProcessor.singleton.reportNavigation(path: path)
+
+        return modifier(EmptyModifier())
+    }
+
+    public func instrumentNavigation(path: Encodable) -> some View {
+        HoneycombNavigationProcessor.singleton.reportNavigation(path: path)
+
+        return modifier(EmptyModifier())
+    }
+}
+
+public struct HoneycombNavigationPathSpanProcessor: SpanProcessor {
+    public let isStartRequired = true
+    public let isEndRequired = false
+
+    public func onStart(
+        parentContext: SpanContext?,
+        span: any ReadableSpan
+    ) {
+        if HoneycombNavigationProcessor.singleton.currentNavigationPath != nil {
+            span.setAttribute(
+                key: "CurrentNavigationPath",
```

Comment:
In Android, they call this attribute `screen.name`. I think it would be good to be consistent with that, especially for use in Timeline Analysis.

https://github.com/open-telemetry/opentelemetry-android/blob/c10adbaf11dbc415190d5fdf8dcf5183485d3ef6/core/src/main/java/io/opentelemetry/android/common/RumConstants.java#L20

Reply:
sounds good. Do you want me to change it on the Navigation spans as well?

Reply:
yeah, I think so